src/main/typescript/actor/wvActor.ts

```diff
@@ -476,58 +482,82 @@ export default class WvActor extends Actor {
     });
   }
 
+  async rollCreateCheckMessage(
+    flavor: string,
+    baseFormula: Formulator,
+    target: SerializedCompositeNumber,
+    options: RollOptions | undefined
+  ): Promise<void> {
+    const criticals = this.data.data.secondary.criticals;
+    const fullFormula = baseFormula.modify(options?.modifier).criticals({
+      success: criticals.success.total,
+      failure: criticals.failure.total
+    });
+    const checkRoll = new Roll(fullFormula.toString()).roll({ async: false });
+
+    const msgOptions = createDefaultMessageData(
+      ChatMessage.getSpeaker({ actor: this }),
+      options?.rollMode ?? getGame().settings.get("core", "rollMode")
+    );
+
+    msgOptions.flavor = flavor;
+
+    await diceSoNice(checkRoll, msgOptions.whisper, { actor: this.id });
```

Comment:
This might also need a check for blind rolls. Right now I think a user would see the 3D dice roll animation if the roll was blind.